tests/pipelines/base-unit-validate.yml

```diff
@@ -0,0 +1,220 @@
+name: "ALZ-Bicep - Unit Tests - Validate All Modules"
+
+trigger: none
+
+variables:
+  - group: csu-bicep-environment
+  - name: ResourceGroupName
+    value: "rsg-github-pr-$(System.PullRequest.PullRequestNumber)"
+  - name: ManagementGroupPrefix
+    value: "PR-$(System.PullRequest.PullRequestNumber)"
+  - name: TopLevelManagementGroupDisplayName
+    value: "PR $(System.PullRequest.PullRequestNumber) Azure Landing Zones"
+  - name: SubscriptionName
+    value: "sub-unit-test-pr-$(System.PullRequest.PullRequestNumber)"
+
+jobs:
+- job: bicep_validate
+  displayName: Validate Bicep Files for PR
+  pool: 
+    vmImage: ubuntu-latest
+  steps:
+  - task: Bash@3
+    displayName: Login to Azure
+    name: git_azlogin
+    inputs:
+      targetType: 'inline'
+      script: |
+        az login --service-principal --username $(azclilogin) --password $(azclipwd) --tenant $(azclitenant)
+
+  - task: Bash@3
+    displayName: Az CLI Create Subscription for PR
+    name: create_subscription
+    inputs:
+      targetType: 'inline'
+      script: |
+        subid=$(az deployment tenant create --name "deploy-$(SubscriptionName)" --location $(Location) --template-file infra-as-code/bicep/CRML/subscriptionAlias/subscriptionAlias.bicep --parameters @infra-as-code/bicep/CRML/subscriptionAlias/parameters/subscriptionAlias.parameters.min.json --parameters parSubscriptionBillingScope=$(ALZ-AZURE-SECRET-EA-BILLING-ACCOUNT) parSubscriptionName=$(SubscriptionName) | jq .properties.outputs.outSubscriptionId.value | tr -d '"')
+        echo $subId
+        echo "##vso[task.setvariable variable=subscriptionId]$subid"
+        
+  - task: Bash@3
+    displayName: Az CLI Refresh subscription list
+    name: refresh_subscription
+    inputs:
+      targetType: 'inline'
+      script: |
+        az account list --refresh        
+        
+  - task: Bash@3    
+    displayName: Az CLI Create Resource Group for PR
+    name: create_rsg
+    inputs:
+      targetType: 'inline'
+      script: |
+        az account set --subscription $(subscriptionId)
+        if [ $(az group exists --name $(ResourceGroupName) ) == false ]; then
+            sleep 300
+        fi
+        az deployment sub create --name "deploy-$(ResourceGroupName)" --template-file infra-as-code/bicep/modules/resourceGroup/resourceGroup.bicep --location $(Location) --parameters parLocation=$(Location) parResourceGroupName=$(ResourceGroupName)
+
+  - task: Bash@3     
```

Comment:
Should we also include a validate of this one for completeness? Even though we deploy it i think it would be good to stay consistent and validate first too!

Reply:
We could. I don't see the point, since we are doing the actual deployment regardless though?

Reply:
Your call, happy either way👍👍

Reply:
Let's not then. Going forward we'll have other test pipelines as we discussed, we can put it in there.